src/components/Search/SearchAutocompleteList.tsx

```diff
@@ -652,7 +667,23 @@ function SearchAutocompleteList(
                 return;
             }
 
-            const trimmedUserSearchQuery = getQueryWithoutAutocompletedPart(autocompleteQueryValue);
+            const fieldKey = focusedItem.mapKey?.includes(':') ? focusedItem.mapKey.split(':').at(0) : focusedItem.mapKey;
+            const isNameField = fieldKey && CONTINUATION_DETECTION_SEARCH_FILTER_KEYS.includes(fieldKey as SearchFilterKey);
+
+            let trimmedUserSearchQuery;
+            if (isNameField && fieldKey) {
```

Comment:
We use this logic 3x, in only very slightly different ways; do you think we can strip it out into a helper function? Here and in the two files below.

Reply:
yeah i guess so, it's different enough. Okay this works for me!